### PR TITLE
#702 removing temporary files in tests

### PR DESCRIPTION
**Pull request by krzyk**
For #702 
Replaced all temporary folder/files creation in tests with TemporaryFolders automatic removal of temporary data.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#779](https://github.com/yegor256/cactoos/issues/779) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#779](https://github.com/yegor256/cactoos/issues/779) is assigned to @sharpler[/z](https://www.0crat.com/u/sharpler), here is [why](https://www.0crat.com/footprint/C63314D6Z/1102120463); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

**Comment by krzyk** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/779#issuecomment-383266106))

@krzyk This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#779](https://github.com/yegor256/cactoos/issues/779) is [in scope](http://www.zerocracy.com/policy.html#14) for 7hr
  * The role is `REV`
  * The job is assigned to @sharpler[/z](https://www.0crat.com/u/sharpler) for [7hr](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @sharpler[/z](https://www.0crat.com/u/sharpler) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @krzyk[/z](https://www.0crat.com/u/krzyk): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%23779%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by krzyk** [comment]
@SharplEr ping :)

**Comment by 0crat** [comment]
@sharpler[/z](https://www.0crat.com/u/sharpler) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/test/java/org/cactoos/io/InputStreamOfTest.java [hunk 1]
```diff
@@ -46,10 +48,16 @@
  */
 @SuppressWarnings("PMD.TooManyMethods")
 public final class InputStreamOfTest {
```
Comment (SharplEr) [inline src/test/java/org/cactoos/io/InputStreamOfTest.java]: @krzyk  You have missed in this class usage of `File.createTempFile` in methods: `readsFileContent`, `readsFromUri`, `readsFromUrl`, `readsFromTextWithCharset`.

Comment (krzyk) [reply src/test/java/org/cactoos/io/InputStreamOfTest.java]: @SharplEr Good catch, fixed.

**Comment by SharplEr** [comment]
@krzyk I think it will be nice to add [pdd](http://www.yegor256.com/2010/03/04/pdd.html) todo task for adding `File.createTempFile` to forbidden API as in issue #704

**Comment by SharplEr** [comment]
@krzyk In the project we have class `TempFile`, which use `File.createTempFile` inside. And `TempFile` is being using in tests: `TeeOutputTest`, `WriterAsOutputStreamTest`, `ReaderAsBytesTest` and `GzipOutputTest`. Looks like you can replace it too.

**Comment by krzyk** [comment]
@SharplEr Updated the code, please see again.

**Comment by krzyk** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/779#issuecomment-384643210))

@krzyk This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#779](https://github.com/yegor256/cactoos/issues/779) is [in scope](http://www.zerocracy.com/policy.html#14) for 6days
  * The role is `REV`
  * The job is assigned to @sharpler[/z](https://www.0crat.com/u/sharpler) for [6days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @sharpler[/z](https://www.0crat.com/u/sharpler) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @krzyk[/z](https://www.0crat.com/u/krzyk): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%23779%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by SharplEr** [comment]
@krzyk  Okay, hope we didn't miss anything.
@llorllale  Good to merge.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @sharpler[/z](https://www.0crat.com/u/sharpler), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#779](https://github.com/yegor256/cactoos/issues/779) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @sharpler[/z](https://www.0crat.com/u/sharpler)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**